spring-data-relational/src/main/java/org/springframework/data/relational/core/conversion/DbAction.java

```diff
@@ -423,6 +423,18 @@ public BatchDelete(List<Delete<T>> actions) {
 		}
 	}
 
+	final class BatchDeleteRootWithVersion<T> extends BatchWithValue<T, DeleteRoot<T>, Number> {
+		public BatchDeleteRootWithVersion(List<DeleteRoot<T>> actions) {
+			super(actions, DeleteRoot::getPreviousVersion);
```

Comment:
Grouping deletes in batches having the same version doesn't make much sense. 
The versions gets increased by every update, therefore versions of different aggregates will be widely different.

We should only group by with/without version and use a simple `delete where id = and version =` and just use a batch statement, leaving any optimisation to the JDBC driver.

Reply:
That's a good point. I updated the batching implementation to only batch `DeleteRoot` actions without `previousVersion`.